src/holon/rule_engine/repositories/repository_base.py

```diff
@@ -155,3 +144,44 @@ def remove(self, object):
         <CHANGES INTERNAL STATE>
         """
         raise NotImplementedError()
+
+
+def attribute_matches_value(
+    object: object, attribute_name: str, value, comparator: AttributeFilterComparator
+):
+    # throws if attribute doesn't exist
+    attribute = getattr(object, attribute_name)
+
+    if not is_comparable(attribute, value):
+        raise Exception(
+            f"Can't compare {object.__class__} attribute {attribute_name} to value {value}, because the types don't match: {type(attribute)} vs {type(value)}"
+        )
+
+    if comparator.value == AttributeFilterComparator.EQUAL.value:
+        return attribute == value
+
+    if comparator.value == AttributeFilterComparator.LESS_THAN.value:
+        return attribute < value
+
+    if comparator.value == AttributeFilterComparator.GREATER_THAN.value:
+        return attribute > value
+
+    if comparator.value == AttributeFilterComparator.NOT_EQUAL.value:
+        return attribute != value
+
+    raise Exception("unreachable")
+
+
+def is_comparable(x, y) -> bool:
```

Comment:
Dit doet python zelf eigenlijk ook al toch? Ik denk dat we alle arithmetic validity checks aan python willen overlaten

Reply:
`"sadf" == 12` vindt Python goed, maar waarschijnlijk heeft een modelleur dan een fout gemaakt. 

Ik denk wel dat we wat modellen moeten fixen of uitzonderingen toe moeten voegen wanneer we dit op de echte scenario's gaan testen.

Reply:
`{"wasd": 14} == 12` geeft overigens ook geen error

Reply:
Ik denk dat dit inderdaad gewenst gedrag is, maar volgens mij is het nu het belangerijkst om het gedrag _precies_ hetzelfde te houden als in het oude geval zodat we het makkelijk kunnen valideren. 

Dit zou ik dan als een los issue inschieten die we kunnen fixen wanneer we dit nieuwe systeem opgezet hebben en gevalideerd hebben dat het niet afwijkt van het oude systeem.

Reply:
ok, aangepast